tapdb/universe.go

```diff
@@ -244,23 +244,17 @@ func upsertAssetGen(ctx context.Context, db UpsertAssetStore,
 		}
 	}
 
-	genProof := &proof.Proof{}
-	err = genProof.Decode(bytes.NewReader(genesisProof))
```

Comment:
This is a breaking change, the reason why the raw leaf is used is to decouple it from the file format which includes the prefix, length information, etc. As designed, one can iterate backwards through the universe leaves for a given asset lineage, then assemble all those into proof files as is to verify. With this change, they'd need to strip out all the version/prefix information before they were able to do that. 


Otherwise, let's say we update the disk format in the future (we add an MMR commitment or something like that), would we also go to attempt to update all the serialized state transitions as well? 

I get the optimization motivation here though, perhaps we can memoize the encoding? So you call `Encode` and it returns an existing value instead of doing it over and over again. 

In case we do decide to go this direction, we'd also need to update the BIPs.

Reply:
Nvm, chatted offline, this is just about moving from a blob to the struct.